### ISSUE TITLE
Way to call directly without crosshair click?

### ISSUE THREAD

**Issue by emigre459** [Steffanic/streamlit-geolocation#2]
Love this tool, thanks for pulling it together for such a handy use case! One thing I'd love to be able to do is trigger the call for a geolocation data pull simply by calling `streamlit_geolocation()` instead of having that call spawn the crosshair icon that then needs to be clicked by the user (so I can style my own button for them to click directly for the desired find-my-location effect). Is there a straightforward way to do that with your package?

**Comment by Steffanic** [comment]
Thank you for the kind words! I never thought anyone would use this. 😅 Are you asking to choose your own icon, or to extract the geolocation logic? The first one may be straightforward. As I'm using FontAwesome icons, it should be straightforward to add an icon string prop. As for extracting the functionality, it's critical that this is called only by user request. I'd need to know more about what you are trying to do. I'll think on it and take a look when I get to my computer.

**Comment by joao-pompeu** [comment]
I agree with @emigre459. I think I know what he wants and probably it is the same functionality I'm searching for. I need the call search the coordinates the ininterruptly, without the need to press the cross hair button.

**Comment by Steffanic** [comment]
I understand. However, I remember having two issues. One was that the Mozilla docs indicate that you ought to only ask for the user's location through user request. In other words, you shouldn't just grab the user's location willy-nilly. The other had to do with actually calling the JS. I am a bit hazy on the details, but I think that's why I wrapped it in a custom component.

Looking back at my React component, you might be able to: 
1) Remove the button component so the react component is invisible
2) Add a useEffect in the Streamlitgeolocation component that calls the getLocation function

I'm not sure this will work, I don't have the time to test it, and I am wary of putting code out that goes against Mozilla docs. That said, Godspeed!

**Comment by sandernordeide** [comment]
I think that the browser itself will prompt the user to "Allow location" anyway, if it is called directly.

**Comment by ptrkhh** [comment]
> As for extracting the functionality, it's critical that this is called only by user request. I'd need to know more about what you are trying to do. I'll think on it and take a look when I get to my computer.

Hi @Steffanic, first of all thank you for this amazing package!

Currently, I am building an app that polls the live location periodically (think package tracker, Uber, etc.) and therefore I would need to call the function directly over time the app is live, without the user clicking the button every second or so

As @sandernordeide mentioned, accessing the device location would require user's consent in the first place, so I don't think it will be invasive to the user's privacy
